### PR TITLE
Added users options to DBCreateOptions

### PR DESCRIPTION
**Pull request by rashtao**
fixes https://github.com/arangodb/arangodb-java-driver/issues/348

### CODE REVIEW

File: src/test/java/com/arangodb/ArangoDBTest.java [hunk 1]
```diff
@@ -83,105 +83,138 @@ public static void shutdown() {
...
+
+        DatabaseEntity info = arangoDB.db(dbName).getInfo();
+        assertThat(info.getName(), is(dbName));
+
+        assertThat(arangoDB.getUsers().stream().anyMatch(it -> it.getUser().equals("testUser")), is(true));
+
+        ArangoDB arangoDBTestUser = new ArangoDB.Builder()
+                .user("testUser")
+                .password("testPasswd")
+                .build();
```
Comment (joerg84) [inline src/test/java/com/arangodb/ArangoDBTest.java:199]: Should you include extra in the test as well?

File: src/main/java/com/arangodb/model/DatabaseUsersOptions.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+import java.util.Map;
+
+/**
+ * @author Michele Rastelli
+ */
+public class DatabaseUsersOptions {
+
+    private String username;
+    private Map<String, Object> extra;
```
Comment (joerg84) [inline src/main/java/com/arangodb/model/DatabaseUsersOptions.java:31]: I find the name extra a bit unconventional, but don't have better one. So ok

File: src/test/java/com/arangodb/ArangoDBTest.java (hunk 1 above)
Comment (rashtao) [reply src/test/java/com/arangodb/ArangoDBTest.java:207]: Thanks for spotting it, I added an assertion to check it.

File: src/main/java/com/arangodb/model/DatabaseUsersOptions.java (hunk 2 above)
Comment (rashtao) [reply src/main/java/com/arangodb/model/DatabaseUsersOptions.java:31]: For clarity/consistency reasons, we keep the same field names as in the database API.

**Pull request merged**